internal/tasks/schedulertask_test.go

```diff
@@ -140,3 +142,131 @@ func TestSchedulerTask_Run(t *testing.T) {
 	cancel()
 	<-doneCh
 }
+
+func TestSchedulerTask_refreshCache(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	registryMock := mocks.NewMockIRegistry(ctrl)
+	// watcherMock := mocks.NewMockIWatcher(ctrl)
+	brokerstoreMock := mocks2.NewMockIBrokerStore(ctrl)
+	nodeCoreMock := mocks3.NewMockICore(ctrl)
+	topicCoreMock := mocks4.NewMockICore(ctrl)
+	subscriptionCoreMock := mocks7.NewMockICore(ctrl)
+	nodebindingCoreMock := mocks5.NewMockICore(ctrl)
+	schedulerMock := mocks6.NewMockIScheduler(ctrl)
+
+	workerID := uuid.New().String()
+	type fields struct {
+		id               string
+		registry         registry.IRegistry
+		brokerstore      brokerstore.IBrokerStore
+		nodeCore         node.ICore
+		scheduler        scheduler.IScheduler
+		topicCore        topic.ICore
+		nodeBindingCore  nodebinding.ICore
+		subscriptionCore subscription.ICore
+		nodeCache        map[string]*node.Model
+		subCache         map[string]*subscription.Model
+		topicCache       map[string]*topic.Model
+		nodeWatchData    chan *struct{}
+		subWatchData     chan *struct{}
+		topicWatchData   chan *struct{}
+	}
+	type args struct {
+		ctx context.Context
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "Refresh cache successfully",
+			fields: fields{
+				id:               workerID,
+				registry:         registryMock,
+				brokerstore:      brokerstoreMock,
+				nodeCore:         nodeCoreMock,
+				scheduler:        schedulerMock,
+				topicCore:        topicCoreMock,
+				nodeBindingCore:  nodebindingCoreMock,
+				subscriptionCore: subscriptionCoreMock,
+				nodeCache:        make(map[string]*node.Model),
+				subCache:         make(map[string]*subscription.Model),
+				topicCache:       make(map[string]*topic.Model),
+				nodeWatchData:    make(chan *struct{}),
+				subWatchData:     make(chan *struct{}),
+				topicWatchData:   make(chan *struct{}),
+			},
+			args: args{
+				ctx: ctx,
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			sm := &SchedulerTask{
+				id:               tt.fields.id,
+				registry:         tt.fields.registry,
+				brokerstore:      tt.fields.brokerstore,
+				nodeCore:         tt.fields.nodeCore,
+				scheduler:        tt.fields.scheduler,
+				topicCore:        tt.fields.topicCore,
+				nodeBindingCore:  tt.fields.nodeBindingCore,
+				subscriptionCore: tt.fields.subscriptionCore,
+				nodeCache:        tt.fields.nodeCache,
+				subCache:         tt.fields.subCache,
+				topicCache:       tt.fields.topicCache,
+				nodeWatchData:    tt.fields.nodeWatchData,
+				subWatchData:     tt.fields.subWatchData,
+				topicWatchData:   tt.fields.topicWatchData,
+			}
+			dummyTopicModels := []*topic.Model{
```

Comment:
Please use above method

Reply:
not declared in this file but sure will declare it here as well